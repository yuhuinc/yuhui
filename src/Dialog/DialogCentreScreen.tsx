import React from "react";
import styled from "styled-components";
import {
  useDialogState,
  Dialog,
  DialogBackdrop,
  DialogOptions
} from "reakit/Dialog";
import { colors } from "../shared/constants";

interface DialogProps {
  label: string;
  dialog?: DialogOptions;
  [key: string]: any;
}

const StyledBackdrop = styled(DialogBackdrop)`
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  background-color: rgba(0, 0, 0, 0.5);
  position: fixed;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  z-index: 999;
`;

const StyledDialog = styled(Dialog)`
  display: flex;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  justify-content: center;
  align-items: center;
  z-index: 999;
  max-width: 560px;
  max-height: 250px;
  color: #212121;
  background-color: ${colors.WHITE};
  border-radius: 8px;
  border: 1px solid rgba(33, 33, 33, 0.25);
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  padding: 1em;
  max-height: calc(100vh - 56px);
  outline: 0;
`;

export const DialogCentreScreen = ({
  label,
  children,
  dialog,
  ...rest
}: DialogProps) => {
  dialog = dialog || useDialogState();
  return (
    <StyledBackdrop {...dialog}>
      <StyledDialog {...dialog} aria-label={label} {...rest}>
        {children}
      </StyledDialog>
    </StyledBackdrop>
  );
};
