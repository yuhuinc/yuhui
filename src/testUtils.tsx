import React from "react";
import { render } from "@testing-library/react";

import { ContentProvider } from "./Content/Content";

interface IRenderWithContent {
  theme?: any;
  contentNodes?: any;
  lang?: string;
}

export const renderWithContent = ({
  theme = {},
  contentNodes = {},
  lang = "en"
}: IRenderWithContent) => (ui, options = {}) => {
  const ContentWrapper = ({ children }) => {
    const defaultTheme = {
      colors: {
        primary: {
          light: "#d0dbfd",
          dark: "#230cc2"
        }
      }
    };

    return (
      <ContentProvider
        content={{
          theme: { ...defaultTheme, ...theme },
          contentNodes
        }}
        lang={lang}
      >
        {children}
      </ContentProvider>
    );
  };

  return render(ui, { wrapper: ContentWrapper, ...options });
};
