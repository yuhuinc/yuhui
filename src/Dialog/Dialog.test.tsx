import React from "react";
import faker from "faker";
import { DialogCentreScreen } from "./DialogCentreScreen";
import { renderWithContent } from "../testUtils";

describe("DialogCentreScreen content", () => {
  test("renders English copy", () => {
    const enCopy = "snapshot";
    const content = {
      contentNodes: {
        dialog: {
          copy: {
            en: enCopy,
            fr: faker.lorem.word()
          }
        }
      },
      lang: "en"
    };

    const { container } = renderWithContent(content)(
      <DialogCentreScreen label={enCopy}>
        <p>{enCopy}</p>
      </DialogCentreScreen>
    );

    expect(container).toMatchSnapshot();
  });

  test("renders French copy", () => {
    const frCopy = "snapshot";
    const content = {
      contentNodes: {
        dialog: {
          copy: {
            en: faker.lorem.word(),
            fr: frCopy
          }
        }
      },
      lang: "fr"
    };

    const { container } = renderWithContent(content)(
      <DialogCentreScreen label={frCopy}>
        <p>{frCopy}</p>
      </DialogCentreScreen>
    );

    expect(container).toMatchSnapshot();
  });
});
