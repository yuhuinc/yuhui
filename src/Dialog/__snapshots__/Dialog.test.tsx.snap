// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogCentreScreen content renders English copy 1`] = `<div />`;

exports[`DialogCentreScreen content renders French copy 1`] = `<div />`;
